libs/blocks/aside/aside.js

```diff
@@ -46,32 +47,101 @@ function decorateStaticLinks(el) {
   textLinks.forEach((link) => { link.classList.add('static'); });
 }
 
+function decorateMedia(el) {
+  if (!(el.classList.contains('medium') || el.classList.contains('large'))) return;
+  const allMedia = el.querySelectorAll('div > p video, div > p picture');
+  let processed = false;
+  allMedia.forEach((media) => {
+    if (processed) return;
+    const parentP = media.closest('p');
+    const siblingP = parentP?.nextElementSibling;
+    if (!siblingP || siblingP.nodeName.toLowerCase() !== 'p') return;
+    const siblingText = siblingP.textContent;
+    const hasFormats = siblingText.match(FORMAT_REGEX)?.index;
+    if (!(hasFormats === 0)) return;
+    processed = true;
+    const formats = siblingText.split(': ')[1]?.split(/\s+/);
+    const formatClasses = [];
+    if (formats) {
+      formatClasses.push('format');
+      if (formats.length === 3) formatClasses.push(`desktop-${formats[2]}`);
+      if (formats.length >= 2) formatClasses.push(`tablet-${formats[1]}`);
+      formatClasses.push(`mobile-${formats[0]}`);
+    }
+    media.closest('div').classList.add(...formatClasses);
+    siblingP.remove();
+    media.closest('div').insertBefore(media, parentP);
+    parentP.remove();
+  });
+}
+
+function decorateVideo(container) {
+  const link = container.querySelector('a[href*=".mp4"]');
+  if (!link) return;
+  const isNotLooped = link.hash?.includes('autoplay1');
+  const attrs = `playsinline autoplay ${isNotLooped ? '' : 'loop'} muted`;
+  container.innerHTML = `<video preload="metadata" ${attrs}>
+    <source src="${link.href}" type="video/mp4" />
+  </video>`;
+  container.classList.add('has-video');
+}
+
+function decorateBlockBg(block, node) {
+  const viewports = ['mobile-only', 'tablet-only', 'desktop-only'];
+  const childCount = node.childElementCount;
+  const { children } = node;
+  node.classList.add('background');
+  if (childCount === 2) {
+    children[0].classList.add(viewports[0], viewports[1]);
+    children[1].classList.add(viewports[2]);
+  }
+  [...children].forEach(async (child, index) => {
```

Comment:
Nothing async inside the function

Reply:
@chrischrischris I have made the requested changes.